src/server/bg_services/lifecycle.js

```diff
@@ -11,63 +11,113 @@ const system_store = require('../system_services/system_store').get_instance();
 const auth_server = require('../common_services/auth_server');
 
 const LIFECYCLE = {
-    schedule_min: 1 //run every 5 minutes
+    schedule_min: 5 * 1000 * 60 // run every 5 minutes
 };
 
+function get_expiration_timestamp(expiration) {
+    if (!expiration) {
+        return undefined; // undefined
+    } else if (expiration.date) {
+        return Math.floor(new Date(expiration.date).getTime() / 1000);
+    } else if (expiration.days) {
+        return moment().subtract(expiration.days, 'days').unix();
+    }
+}
+
+/*
+*
+* S3 ExpiredObjectDeleteMarker logic
+*        If set to true, the delete marker will be expired; if set to false the policy takes no action.
+*
+* MDStore find_objects() filter_delete_markers logic:
+*       if (filter_delete_markers === true) delete_marker = null;
+*       else if (filter_delete_markers === false) delete_marker = true;
+*/
+function get_expiration_filter_delete_markers(expiration) {
+    if (expiration && expiration.expired_object_delete_marker) {
+        return false; // delete_marker = true
+    }
+    return true; // delete_marker = null
+}
+
+function get_filter_field(filter, field) {
+    return filter ? filter[field] : undefined;
+}
+
+function get_prefix(filter) {
+    return get_filter_field(filter, 'prefix');
+}
+
+function get_size_less_than(filter) {
+    return get_filter_field(filter, 'object_size_less_than');
+}
+
+function get_size_greater_than(filter) {
+    return get_filter_field(filter, 'object_size_greater_than');
+}
+
+function get_tags(filter) {
+    return get_filter_field(filter, 'tags');
+}
+
+async function handle_bucket_rule(system, rule, j, bucket) {
+    dbg.log0('LIFECYCLE HANDLING BUCKET:', bucket.name, '(bucket id:', bucket._id, ') BEGIN');
```

Comment:
you added a lot of log0 messages. I'm not sure if all these log messages should be `dbg.log0`. We should be careful not to make our logs noisier than they already are.

Reply:
👍